docs/source/index.rst

```diff
@@ -56,6 +56,9 @@ NumPyro documentation
    tutorials/bayesian_imputation
    examples/gp
    examples/bnn
+   examples/stein_bnn
+   examples/stein_mixture_dmm
+   examples/stein_2d_toy
```

Comment:
How about moving those examples to a separate section? I just feel that `Applications` is a bit long for now and needs refactoring. :) In Pyro, we have [Other inference algorithms](https://pyro.ai/examples/baseball.html) section which we can mimic and use for SteinVI, nested sampling, hmcecs, MixedHMC, SA,...